### DISCUSSION TITLE
Weight calculations in GUI

### DISCUSSION THREAD

**Discussion by carno44** [Eomys/pyleecan#710] [Q&A]
Hi All,

![Error](https://github.com/user-attachments/assets/3a6eb68a-e3e8-4127-ac98-44cef904da7f)
I have configured a IPMSM machine in Pyleecan GUI and the summary is shown in the attached picture. I have questions on the weight calculation of the components. I am checking the relative weights of the stator, rotor lamination, magnets and shaft. 

1. The rotor lamination weight looks lower in comparison with the stator lamination. Both materials are same.
2. The magnets weight look way higher than the intuitive estimation.
3. Shaft weight is 0.

I want to know, what could be the mistake from my side. I have thoroughly checked the material assignment; if requied, I can attach the screenshots of the same.

Thanks,
Carno

**Comment by BonneelP** [comment]
Hello, 

Thank you for your interest in pyleecan. There is maybe a bug in the mass computation or in the machine definition by the GUI. Can you share your machine .json file so I can take a look ? My guess would be something around the length of the lamination/shaft/magnet that are maybe not set properly.

Best regards,
Pierre

**Reply by carno44** [reply]
Thanks you very much @BonneelP for the reply. Pyleecan is really a great tool and very easy to work on. 

I am attaching the required file for your reference. Also, I simulaed the same machine in MotorAnalysisPM. The results of Pyleecan and MotorAnalysisPM are very different. Even if I consider my not much experience in motor design and simulations, the results should not differ this much. I will attach the efficiency map of the machine by the two tools in next response. 

[IPMSM_431.json](https://github.com/user-attachments/files/17485112/IPMSM_431.json)

Regards,
Carno

**Comment by carno44** [comment]
Hi  @BonneelP, I am still working on the problem without success. Please let me know, if you find something. Thanks in advance. 

Regards,
Carno
